app/instrumentation-tests/src/org/commcare/androidTests/CaseListSearchTest.kt

```diff
@@ -0,0 +1,138 @@
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import org.commcare.dalvik.R
+import org.commcare.utils.CustomMatchers
+import org.commcare.utils.InstrumentationUtility
+import org.hamcrest.Matchers.*
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+class CaseListSearchTest: BaseTest() {
+
+    companion object {
+        const val CCZ_NAME = "test_list_search.ccz"
+        const val APP_NAME = "Test: List Searching"
+        val entities = arrayOf("Christy", "Matthew", "Steven", "Tuco")
+    }
+
+    @Before
+    fun setup() {
+        installApp(APP_NAME, CCZ_NAME)
+        InstrumentationUtility.login("test_list_search", "123")
+    }
+
+    @Test
+    fun testSearch() {
+        gotoSearchTestModule()
+
+        // Confirm 4 items in the list
+        onView(withId(R.id.screen_entity_select_list))
+                .check(matches(CustomMatchers.matchListSize(4)))
+        matchListItems(entities)
+
+        onView(withId(R.id.search_action_bar))
+                .perform(click())
+
+        val searchTextId = Resources.getSystem().getIdentifier("android:id/search_src_text", null, null)
+
+        // type `c` and search
+        onView(withClassName(endsWith("SearchAutoComplete")))
+                .perform(typeText("c"))
+        matchList(2, 4, "c", arrayOf(entities[0], entities[3]))
+
+        // type `h` and search
+        onView(withClassName(endsWith("SearchAutoComplete")))
+                .perform(typeText("h"))
+        matchList(1, 4, "ch", arrayOf(entities[0]))
+
+        // Confirm it persists on rotation
+        InstrumentationUtility.rotateLeft()
+        Espresso.closeSoftKeyboard()
+        matchList(1, 4, "ch", arrayOf(entities[0]))
+
+        InstrumentationUtility.rotatePortrait()
+        Espresso.closeSoftKeyboard()
+        matchList(1, 4, "ch", arrayOf(entities[0]))
+
+        onView(allOf(withClassName(endsWith("ImageView")), withContentDescription("Clear query")))
+                .perform(click())
+        onView(withText(containsString("results for your search")))
+                .check(matches(not(isDisplayed())))
+        matchListItems(entities)
+
+        onView(withText("Search Tests"))
+                .check(doesNotExist())
+        InstrumentationUtility.gotoHome()
+
+        // Start Over
+        gotoSearchTestModule()
+
+        // Clearing searches the other way
+        onView(withId(R.id.search_action_bar))
+                .perform(click())
+        onView(withClassName(endsWith("SearchAutoComplete")))
+                .perform(typeText("x"))
+        matchList(0, 4, "x", null)
+        closeSoftKeyboard()
+        // Use back to clear search
+        // For some weird reasons, pressing back once doesn't work with searchview.
```

Comment:
Actually the first back press there is to hide the keyboard, which I'm already doing here using `closeSoftKeyboard()`. 
I added this comment because it was [working earlier](https://app-automate.browserstack.com/dashboard/v2/builds/36093559468ebcd7fafd8d88d09c7558f74ceed9/sessions/tests/57461310417665873c6a7add71fcdd02b15d03b08d74f402) but after rebasing with master it isn't anymore.

Reply:
Or maybe `closeSoftKeyboard()` doesn't remove focus from search edit text which is again something related with `searchview` only.

Reply:
@shubham1g5 bumping you for your thoughts on this.

Reply:
@ShivamPokhriyal My point was just that if we know the reason for this behaviour we should state that as comment instead of being ambiguous. You are the best person in capacity to validate your reasoning here and append the comment accordingly, so I will let you decide on any action item on this. This is not a big deal and PR is approved from my side.

Reply:
Cool, I'll keep it as is then.